azurelinuxagent/common/utils/shellutil.py

```diff
@@ -57,6 +57,11 @@ def __str__(self):
 Shell command util functions
 """
 
+def has_command(cmd):
+    """
+    Return True if the given command is on the path
+    """
+    return not run(cmd, False)
```

Comment:
but the design here is a bit different, the distro itself is not abstracted, meaning I cannot ask distro.what_is_your_route_command. The abstraction chose in the agent implementation is distro_name.py whereever specialization may be needed and that gets us back to what I stated above, an obsolete implementation that uses "route" and lost of identical files that use "ip route"

Reply:
When the numbers show that most distros use "ip route", we'll change the default to use that.

If you _had to_ do runtime discovery of the presence of a command, you'd want to persist that information so that subsequent invocations of "do this routing thing" wouldn't be probing the system over and over again. (But I don't think you have to do this, for this purpose, at this time.)

Reply:
> When the numbers show that most distros use "ip route", we'll change the default to use that.

AFAIK all our endorsed distro images and versions have iproute2 utilities installed already, please correct me if I'm wrong.  Most might also have the obsolete route/ifconfig/etc. tools as well, but that's no reason to keep them as default forever.

Reply:
@rjschwei are you willing to make the change to iproute2? @boumenot what are your thoughts?